src/components/tabs/tab.tsx

```diff
@@ -55,12 +50,10 @@ export const EuiTab: FunctionComponent<Props> = ({
       <a
         role="tab"
         aria-selected={!!isSelected}
-        onClick={onClick as MouseEventHandler<HTMLAnchorElement>}
```

Comment:
Just a clean up to avoid the ominous `as` casting. `onClick` is now part of `...rest`, which is more of a subjective choice, but it's more in line with how we've handled other button-like components.